### PR TITLE
Working startWith operator - Firestore

### PR DESCRIPTION
**Pull request by MXPOL**
Turns out that the `startWith` operator did not work properly, This PR fixed it

### CODE REVIEW

File: packages/external-db-firestore/lib/sql_filter_transformer.js [hunk 1]
```diff
@@ -130,6 +134,19 @@ class FilterParser {
...
+    parseStringBegins(fieldName, inlineFields, value) {
+        return [{
+            fieldName: this.inlineVariableIfNeeded(fieldName, inlineFields),
+            opStr: '>=',
+            value,
+        },
+        {
+            fieldName: this.inlineVariableIfNeeded(fieldName, inlineFields),
+            opStr: '<',
+            value: value + String.fromCharCode(65535)
```
Comment (Idokah) [inline packages/external-db-firestore/lib/sql_filter_transformer.js:146]: `String.fromCharCode(65535)` should be at firestore_utils as `MaxChar` in my opinion

Comment (noam-almog) [reply packages/external-db-firestore/lib/sql_filter_transformer.js:146]: pack this value to a function or const with proper name so we can understand what you did there.

File: packages/external-db-firestore/lib/firestore_utils.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 const { Timestamp } = require('@google-cloud/firestore')
 
+const lastLetterCode = String.fromCharCode(65535)
```
Comment (Idokah) [inline packages/external-db-firestore/lib/firestore_utils.js:3]: That's constant - I would rename it to `LastLetterCoder`

File: packages/external-db-firestore/lib/sql_filter_transformer.js [hunk 3]
```diff
@@ -49,9 +54,13 @@ class FilterParser {
     isSingleFieldOperator(operator) {
         return [ne, lt, lte, gt, gte, include, eq, string_begins, string_ends].includes(operator)
     }
+
+    isStringBeginsOperator(operator) {
+        return operator === string_begins
+    }
```
Comment (Idokah) [inline packages/external-db-firestore/lib/sql_filter_transformer.js:60]: I think you can check this condition in the if statement.

Comment (MXPOL) [reply packages/external-db-firestore/lib/sql_filter_transformer.js:60]: I prefer it to be clear what we are actually checking

**Pull request merged**